envoy/api/v2/discovery.proto

```diff
@@ -36,6 +36,10 @@ message DiscoveryRequest {
   // explicitly enumerated in resource_names.
   repeated string resource_names = 3;
 
+  // When true the list of resource_names is added to the set of tracked resources
```

Comment:
Another question to consider: how do we know if a cluster can't be obtained from the management server? E.g. if we get some header, the filter decides it wants a service referenced in the header, but the xDS server doesn't know about it.

Reply:
I suppose if the server doesn't respond with the requested cluster and a higher nonce, Envoy can assume it does not exist, but that seems very fragile. Perhaps we should be adding some ability to carry back error information from the management server? I could imagine the management server responding with the name of a requested resource and an associated error?